googlesource/googlesource.go

```diff
@@ -46,6 +46,23 @@ func (client *client) Client() (*bigquery.Client, error) {
 	return bigquery.NewClient(client.ctx, client.projectID, client.opts...)
 }
 
+type bqClient interface {
+	Query(q string) *bigquery.Query
```

Comment:
Why not return results here?

Reply:
this is the same function which BigQuery endpoint exposes. And I had to keep the implementation same.

Reply:
I think that's a good step forward. However, with returning just the query, it won't be easy to mock the results in a unit test. E.g. what if we want to empty results, or paging, or an error being returned from BQ? 

In other words, what I'm suggesting is to have a client which will execute a string query (hiding all the details from us), and not only construct a Query struct.

Reply:
We need to mock iterator and its function in that case. Because if I mock the query also then too Next function implementation would remain same.

Reply:
I am talking about - https://github.com/neha-Gupta1/conduit-connector-bigquery/blob/8e60308d86ed0c8ad803a5a11dc1954eb19b7fce/googlesource/googlesource.go#L122 'Next()' function

Reply:
Since I think we cannot mock the BQ RowIterator (since it's a struct?), we can have our own RowIterator interface. The production code will use the real BQ RowIterator, the test code can mock it, and we can return anything we want in the tests.

Reply:
Done